Common/Data/Custom/Tiingo/TiingoDailyData.cs

```diff
@@ -0,0 +1,189 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using QuantConnect.Data.UniverseSelection;
+
+namespace QuantConnect.Data.Custom.Tiingo
+{
+    /// <summary>
+    /// Tiingo daily price data
+    /// https://api.tiingo.com/docs/tiingo/daily
+    /// </summary>
+    public class TiingoDailyData : BaseData
+    {
+        private static string _authToken = string.Empty;
+        private readonly ConcurrentDictionary<string, DateTime> _startDates = new ConcurrentDictionary<string, DateTime>();
+
+        /// <summary>
+        /// Returns true if the Tiingo API token has been set.
+        /// </summary>
+        public static bool IsAuthTokenSet { get; private set; }
+
+        /// <summary>
+        /// Sets the Tiingo API token.
+        /// </summary>
+        /// <param name="authToken">The Tiingo API token</param>
+        public static void SetAuthToken(string authToken)
```

Comment:
To keep naming convention same as Quandl please use "SetAuthCode" or provide an overload which passes through here as SetAuthToken.